swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+
+channels:
+
+  dataset/pull/request:
+    subscribe:
+      operationId: odf.dataset.pull.request
+      summary: Initiates dataset pull from the ODF server
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPullRequest
+        name: DatasetPullRequest
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - knownHead          
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            knownHead:
+              type: string
+              format: multihash
+              description: Hash of the metadata head block known to the consumer
+
+  dataset/pull/response:
+    publish:
+      operationId: odf.dataset.pull.response
+      summary: Indicates succesfull response to dataset pull request
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPullResponse
+        name: DatasetPullResponse
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - prevKnownHead
+            - newBlocks
+            - dataSlices
+            - checkpoints
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            prevKnownHead:
+              type: string
+              format: multihash
+              description: Hash of the metadata head block that was previously known to consumer
+            newBlocks:
+              $ref: '#/components/schemas/TarOfBlocks'
+            dataSlices:
+              type: array
+              description: List of data slice file hashes referenced by the new blocks
+              items:
+                type: string
+                format: multihash
+            checkpoints:
+              type: array
+              description: List of checkpoint file hashes referenced by the new blocks
+              items:
+                type: string
+                format: multihash
+  
+  dataset/pull/data/negotiate:
+    subscribe:
+      operationId: odf.dataset.pull.data.negotiate
+      summary: Requests to negotiate data download protocol for 1 or more files
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPullDataNegotiate
+        name: DatasetPullDataNegotiate
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - dataFiles
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier     
+            dataFiles:
```

Comment:
"dataFiles" confused me as I immediately thought "what about checkpoints?".

Suggested below to call them all "objects" - also in case we'll expand protocol later with more file types.

Reply:
Already addressed, calling `objectFiles`